kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/api/common/NamespacedRef.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.api.common;
```

Comment:
If you think about it `NamespacedRef` shouldn't be in the `api.common` package, because it should never be used in a CRD (that's the role of the `GlobalRef` I mentioned yesterday). This one is an internal detail of the operator. So let's move it there. We'll have to get rid of `LocalRef.inNamespaceOf()`, but it's just as easily made a static factory method of this class, i.e. `NamespacedRef.namespaced(LocalRef<T> ref, HasMetadata resource)`.

Reply:
yep can do.